kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1352 @@
+    * @param rTransformationMatrix The local transformation contribution
+    * @param rSlaveEquationId The equation id of the slave dofs
+    * @param rMasterEquationId The equation id of the master dofs
+    * @param rLockArray The lock of the dof
+    */
+    void AssembleRelationMatrix(
+        TSystemMatrixType& rT,
+        const LocalSystemMatrixType& rTransformationMatrix,
+        const EquationIdVectorType& rSlaveEquationId,
+        const EquationIdVectorType& rMasterEquationId
+    #ifdef _OPENMP
+        ,std::vector< omp_lock_t >& rLockArray
+    #endif
+        )
+    {
+        SizeType local_size_1 = rTransformationMatrix.size1();
+        SizeType local_size_2 = rTransformationMatrix.size2();
+
+        for (IndexType i_local = 0; i_local < local_size_1; ++i_local) {
+            IndexType i_global = rSlaveEquationId[i_local];
+
+            if (i_global < BaseType::mEquationSystemSize) {
+            #ifdef _OPENMP
+                omp_set_lock(&rLockArray[i_global]);
+            #endif
+                for (IndexType j_local = 0; j_local < local_size_2; ++j_local) {
+                    IndexType j_global = mSolvableDoFReorder[rMasterEquationId[j_local]];
+                    if (j_global < BaseType::mEquationSystemSize) {
+                        rT(i_global, j_global) += rTransformationMatrix(i_local, j_local);
+                    }
+                }
+            #ifdef _OPENMP
+                omp_unset_lock(&rLockArray[i_global]);
+            #endif
+
+            }
+            //note that assembly on fixed rows is not performed here
+        }
+    }
+
+    /**
+     * @brief This method construcs the relationship between the DoF
+     * @param pScheme The integration scheme
+     * @param rA The LHS of the system
+     * @param rModelPart The model part which defines the problem
+     */
+    void ConstructMatrixStructure(
+        typename TSchemeType::Pointer pScheme,
+        TSystemMatrixType& rA,
+        ModelPart& rModelPart
+        ) override
+    {
+        if(rModelPart.MasterSlaveConstraints().size() > 0)
+            ConstructMatrixStructureWithConstraints(pScheme, rA, rModelPart);
+        else
+            BaseType::ConstructMatrixStructure(pScheme, rA, rModelPart);
+    }
+
+    /**
+     * @brief The same methods as the base class but with constraints
+     * @param pScheme The pointer to the integration scheme
+     * @param rModelPart The model part to compute
+     * @param rA The LHS matrix of the system of equations
+     * @param rDx The vector of unkowns
+     * @param rb The RHS vector of the system of equations
+     */
+    void BuildAndSolveWithConstraints(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        )
+    {
+        KRATOS_TRY
+
+        Timer::Start("Build");
+
+        Build(pScheme, rModelPart, rA, rb);
```

Comment:
Did u apply T here? if so, please create a Function BuildWithConstraints and pass to and from it T, so not to have to reform it again.
In any case...how can u use the same Build function as usually??

Reply:
I do that, check it please

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/b31633b342ef4b1411bea2bafc923add598b6477/kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h#L995

Reply:
The entire point is that if you have T than you can obtain the update for the slave dofs simply by an spmv product

Reply:
What i am telling is to return T from here

Reply:
When you call build it does as the block_builder_and_solver_with_constraints. If there are not constraints it does a regular Build, otherwise it does a BuildWithConstraints.

T is a member variable of the builder and solver (then is assembled once, or each time the builder and solver is cleared)